Algorithm/SINGON/11054.py

```diff
@@ -0,0 +1,29 @@
+import sys
+input = sys.stdin.readline
+def solution(N,A):
+    if N==1 or N==2: #Trivial
+        return N
+    dp = [0]*1001
+    dn = [0]*1001
+    for i in range(1,N+1):
+        maximum = 0
+        for j in range(i):
+            if A[j] < A[i]:
+                if dp[j] >= maximum:
+                    maximum = dp[j]
+        dp[i] = maximum+1 #i 번째 까지의 증가수열의 최대 길이
+    for i in range(N,0,-1):
+        reverse_maximum = 0
+        for j in range(N,i-1,-1):
+            if A[j] < A[i]:
+                if dn[j] >= reverse_maximum:
+                    reverse_maximum = dn[j]
+        dn[i] = reverse_maximum+1 #i 번째부터 N까지의 감소함수의 최대 길이
+    ds = [0]*1001
+    for i in range(0,N+1):
+        ds[i] = dp[i]+dn[i]
+    return(max(ds)-1) #i번째 길이를 셀 때 두 번 더했으므로 -1이 필수!
```

Comment:
```suggestion
    return max([p+n for p, n in zip(dp, dn)])-1
```
list comprehension을 추천드립니다 😊

Reply:
오 제가 zip을 잘 이용하지 않고 길게 늘어쓴 것 같습니다. 상민님이 피드백 해주신대로 최대한 list comprehension 쓰도록 노력해보겠습니다. 감사합니다.ㅎㅎ